Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -2647,3 +2647,106 @@ function Remove-CustomRole {
     }
 }
 
+Function Get-vSANDataInTransitEncryptionStatus {
+    <#
+    .DESCRIPTION
+        Gets status of vSAN Data-In-Transit Encryption for all clusters in a SDDC
+    #>
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    param()
+    begin{}
+    process {
+        $clusters = Get-Cluster        
+        $diteConfig = @()
+        $vSANConigView = Get-VsanView -Id VsanVcClusterConfigSystem-vsan-cluster-config-system
+        foreach ($cluster in $clusters) {
+            $diteConfig += [PSCustomObject]@{
+                Name = $cluster.Name
+                DataEncryptionInTransit = $vSANConigView.VsanClusterGetConfig($cluster.ExtensionData.MoRef).DataInTransitEncryptionConfig.Enabled
+            }
+        }
+        $diteConfig | Format-Table | Out-String | Write-Host
+    }
+
+}
+
+Function Set-vSANDataInTransitEncryption {
+    <#
+    .DESCRIPTION
+        Enable/Disable vSAN Data-In-Transit Encryption for clusters of a SDDC
+    .PARAMETER ClusterName
+        Name of the cluster. Leave blank if required to enable for whole SDDC else enter comma separated list of names.
+    .PARAMETER Enable
+        Specify True/False to Enable/Disable the feature.
+    #>
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $false)]
+        [string]
+        $ClusterName,
+        [Parameter(Mandatory = $true)]
+        [bool]
+        $Enable
+    )
+    begin {
+        If ([string]::IsNullOrEmpty($ClusterName)){}
+        Else {
+            $ClusterNamesParsed = Limit-WildcardsandCodeInjectionCharacters -String $ClusterName
+            $ClusterNamesArray = Convert-StringToArray -String $ClusterNamesParsed
+        }
+        Write-Host "Enable value is $Enable"
+        $TagName = "vSAN Data-In-Transit Encryption"  
+        $InfoMessage = "Info - There may be a performance impact when vSAN Data-In-Transit Encryption is enabled. Refer :  https://blogs.vmware.com/virtualblocks/2021/08/12/storageminute-vsan-data-encryption-performance/"      
+    }
+    process {
+        If ([string]::IsNullOrEmpty($ClusterNamesArray)) {
```

Comment:
(D)ont (R)epeat (Y)ourself:
it would seem the body of code inside the foreach is the same in both branches, see if you can separate obtaining the list of clusters from how the list is processed. We have examples of that all over the code here.